types/module/module.go

```diff
@@ -346,11 +346,19 @@ func (m Manager) RunMigrations(ctx sdk.Context, cfg Configurator, fromVM Version
 
 	updatedVM := make(VersionMap)
 	for moduleName, module := range m.Modules {
-		err := c.runModuleMigrations(ctx, moduleName, fromVM[moduleName], module.ConsensusVersion())
-		updatedVM[moduleName] = module.ConsensusVersion()
-		if err != nil {
-			return nil, err
+		fromVersion := fromVM[moduleName]
+		toVersion := module.ConsensusVersion()
+
+		// only run migrations when the from version is > 0
+		// from version will be 0 when a new module is added and migrations shouldn't be run in this case
+		if fromVersion > 0 {
```

Comment:
I'm thinking about the hub right now. It has fromVersion == 0 for all modules.

I agree with skipping & calling InitGenesis (with an empty genesis) for new modules with fromVersion == 0. However, for the hub with pre-existing modules with fromversion == 0, we should do something too.

The easiest way I can think of is add a hardcoded map of old module versions inside the UpgradeHandler

Reply:
Yep I see no other choice if this is the first upgrade. We need to make sure this is well documented a.